src/Nowruz/pages/jobs/detail/applicants/useApplicantAction.tsx

```diff
@@ -20,17 +30,55 @@ export const useApplicantAction = (
   const [columnVisibility, setColumnVisibility] = useState({
     select: false,
   });
-
+  const [searchTerm, setSearchTerm] = useState('');
+  const [pages, setPages] = useState([
+    { tab: 'applicants', page: 1 },
+    { tab: 'offered', page: 1 },
+    { tab: 'rejected', page: 1 },
+  ]);
+  const [totalCounts, setTotalCounts] = useState([
+    { tab: 'applicants', count: applicants.total_count },
+    { tab: 'offered', count: applicants.total_count },
+    { tab: 'rejected', count: applicants.total_count },
+  ]);
   const currentSelectedId = useRef<string>();
 
   const navigate = useNavigate();
 
   useEffect(() => {
+    setSearchTerm('');
     setColumnVisibility({ select: currentTab === 'applicants' });
+    setApplicantsList(applicants.items);
   }, [currentTab]);
 
+  const searchApplicants = async () => {
+    try {
+      const currentPage = pages.find(item => item.tab === currentTab);
+      const status = currentTab === 'applicants' ? 'PENDING' : currentTab === 'offered' ? 'OFFERED' : 'REJECTED';
+      const res = await search(
+        { type: 'applicants', q: searchTerm, filter: { project_id: jobId, status: status } },
+        { page: currentPage?.page, limit: PER_PAGE },
+      );
+      setApplicantsList(res.items);
+      setTotalCounts(totalCounts =>
+        totalCounts.map(item => (item.tab === currentTab ? { ...item, count: res.total_count } : item)),
+      );
+    } catch (e) {
+      // FIXME: Use a global error handling approach
+      console.log('error in search applicants', e);
+    }
+  };
+
+  useEffect(() => {
+    handleChangePage(1);
+  }, [searchTerm]);
+
+  useEffect(() => {
+    searchApplicants();
```

Comment:
could you handle it without `useEffect`?

Reply:
This is for handling pagination and it is the same approach that we have used in all pages. Let's keep it.